### PR TITLE
overhaul burrow charts

### PR DESCRIPTION
**Pull request by gregdhill**
Signed-off-by: Gregory Hill <greg.hill@monax.io>

### CODE REVIEW

File: stable/burrow/values.yaml [hunk 1]
```diff
@@ -1,66 +1,103 @@
...
 chain:
-  name: simpleTestChain
-  id: simpleTestChain
+  name: ""
   numberOfNodes: 1
   logLevel: info
   extraSeeds: []
+  testing: false
+
+validatorAddresses:
```
Comment (compleatang) [inline stable/burrow/values.yaml:12]: let's comment these lines out and add a new file `ci/base.yaml` which includes test addresses and keys it should also flip the `chain.testing` variable to `true`

File: stable/burrow/templates/service-peers.yaml [hunk 2]
```diff
@@ -1,25 +1,36 @@
+{{- $peerIngress := and .Values.peer.ingress.enabled (not (eq (len .Values.peer.ingress.hosts) 0)) -}}
 {{ range (until (.Values.chain.numberOfNodes | int)) }}
 {{- $nodeNumber := printf "%03d" . }}
 ---
 kind: Service
 apiVersion: v1
 metadata:
+{{- if $peerIngress }}
+  annotations:
+    external-dns.alpha.kubernetes.io/hostname: "{{ $nodeNumber }}.{{ index $.Values.peer.ingress.hosts 0 }}"
```
Comment (compleatang) [inline stable/burrow/templates/service-peers.yaml:10]: I'm wondering if we're too opinionated here? Probably not, but maybe something to consider.

File: stable/burrow/values.yaml [hunk 3]
```diff
@@ -1,66 +1,103 @@
...
-  rpc:
-    port: 46657
-    type: "ClusterIP"
-    loadBalance: false
-    node: "000"
-  api:
-    port: 1337
-    type: "ClusterIP"
-    loadBalance: true
+organization: "monax"
```
Comment (compleatang) [inline stable/burrow/values.yaml]: redo this line.

File: stable/burrow/values.yaml [hunk 4]
```diff
@@ -1,66 +1,103 @@
...
+rpcInfo:
+  enabled: true
+  service:
+    port: 26658
+    type: ClusterIP
+    loadBalance: true
+  ingress:
+    enabled: false
+    # exposing partial ingress only exposes the /accounts and /blocks paths outside the cluster
+    partial: true
```
Comment (compleatang) [inline stable/burrow/values.yaml]: let's flip this to `false` as a default.

File: stable/burrow/values.yaml [hunk 5]
```diff
@@ -1,66 +1,103 @@
...
+  port: 6060
 
 resources:
-  # limits:
-  #  cpu: 500m
-  #  memory: 512Mi
-  # requests:
-  #  cpu: 100m
-  #  memory: 128Mi
+  limits:
```
Comment (compleatang) [inline stable/burrow/values.yaml]: let's comment these out by default.

File: stable/burrow/README.md [hunk 6]
```diff
@@ -1,164 +1,183 @@
...
+### Prerequisites
 
-```console
-$ helm install stable/burrow --name my-release
+To deploy a new blockchain network, this chart requires that two objects be present in the same Kubernetes namespace: a configmap should house the genesis file and a secret should hold any validator keys. The provided script, `initialize.sh` automatically provisions a number of files using the [burrow](https://github.com/hyperledger/burrow) toolkit, so please first ensure that `burrow --version` matches the `image.tag` in the [configuration](#configuration). This sequence also requires the very fine [jq](https://stedolan.github.io/jq/) binary to be installed. Two files will be generated, the first of note is `chain-info.yaml` which contains the two necessary Kubernetes specifications to be added to the cluster:
+
+```bash
+CHAIN_NODES=4
+CHAIN_NAME="my-chain-name"
+./initialize.sh
```
Comment (compleatang) [inline stable/burrow/README.md]: let's do it this way...

```bash
curl -LO https://raw.githubusercontent.com/helm/charts/master/initialize.sh
CHAIN_NODES=4 CHAIN_NAME="my-chain-name" ./initialize.sh
kubectl apply ..
```

File: stable/burrow/README.md [hunk 7]
```diff
@@ -1,164 +1,183 @@
...
 
-## Installing the Chart
+## Installation
 
-To install the chart with the release name `my-release`:
+### Prerequisites
 
-```console
-$ helm install stable/burrow --name my-release
+To deploy a new blockchain network, this chart requires that two objects be present in the same Kubernetes namespace: a configmap should house the genesis file and a secret should hold any validator keys. The provided script, `initialize.sh` automatically provisions a number of files using the [burrow](https://github.com/hyperledger/burrow) toolkit, so please first ensure that `burrow --version` matches the `image.tag` in the [configuration](#configuration). This sequence also requires the very fine [jq](https://stedolan.github.io/jq/) binary to be installed. Two files will be generated, the first of note is `chain-info.yaml` which contains the two necessary Kubernetes specifications to be added to the cluster:
```
Comment (compleatang) [inline stable/burrow/README.md]: Let's make it clearer in the second sentence that we need `burrow` and `jq` installed on the operator's environment in order for them to spin up the network.

File: stable/burrow/README.md [hunk 8]
```diff
@@ -1,164 +1,183 @@
...
+
+Another file, `example-init.yaml` contains the the equivalent validator addresses to set in the charts.
+
+### Deployment
+
+To install the chart with the release name `my-release` and the example validator addresses:
+
+```bash
+helm install stable/burrow --name my-release --values example-init.yaml
 ```
```
Comment (compleatang) [inline stable/burrow/README.md:29]: Let's add a comment here that the extra values file is absolutely required for this chart.

File: stable/burrow/README.md [hunk 9]
```diff
@@ -1,164 +1,183 @@
...
+| Parameter | Description | Default |
+| --------- | ----------- | ------- |
+| `image.repository` | image repository | `"hyperledger/burrow"` |
+| `image.tag` | image tag | `"0.23.1"` |
+| `image.pullPolicy` | image pull policy | `"IfNotPresent"` |
+| `chain.name` | human readable name of the blockchain network | `""` |
+| `chain.numberOfNodes` | number of nodes for the blockchain network | `1` |
+| `chain.logLevel` | log level for the nodes (`debug`, `info`, `warn`) | `"info"` |
+| `chain.extraSeeds` | network seeds to dial in addition to the cluster booted by the chart; each entry in the array should be in the form `ip:port` (note: because P2P connects over tcp, the port is absolutely required) | `[]` |
+| `chain.testing` | - | `false` |
```
Comment (compleatang) [inline stable/burrow/README.md]: this is non-standard and we should note here why we use this.

File: stable/burrow/README.md [hunk 10]
```diff
@@ -1,164 +1,183 @@
...
+| `validatorAddresses` | list of validators to deploy | `[]` |
+| `validatorAddresses.Validator_0.Address` | - | `"DEADBEEF"` |
```
Comment (compleatang) [inline stable/burrow/README.md]: empty string these (see comments on values.yaml below)

File: stable/burrow/README.md [hunk 11]
```diff
@@ -1,164 +1,183 @@
...
+| `validatorAddresses.Validator_0.NodeAddress` | - | `"DEADBEEF"` |
+| `env` | environment variables to configure burrow | `{}` |
+| `extraArgs` | extra arguments to give to the build in `burrow start` command | `{}` |
+| `organization` | name of the organization running these nodes (used in the peer's moniker) | `"monax"` |
```
Comment (compleatang) [inline stable/burrow/README.md]: empty string this (see comments on values.yaml below)

File: stable/burrow/README.md [hunk 12]
```diff
@@ -1,164 +1,183 @@
...
+| `rpcGRPC.ingress.enabled` | expose port | `false` |
+| `rpcGRPC.ingress.hosts` | - | `[]` |
+| `rpcGRPC.ingress.annotations` | extra annotations | `` |
+| `rpcGRPC.ingress.tls` | - | `` |
+| `rpcInfo.enabled` | enable RPC service | `true` |
+| `rpcInfo.service.port` | RPC port | `26658` |
+| `rpcInfo.service.type` | service type | `"ClusterIP"` |
+| `rpcInfo.service.loadBalance` | enable load balancing across nodes | `true` |
+| `rpcInfo.ingress.enabled` | expose port | `false` |
+| `rpcInfo.ingress.partial` | exposes the `/accounts` and `/blocks` paths externally | `true` |
```
Comment (compleatang) [inline stable/burrow/README.md]: let's note here that the remainder of the info endpoints are blocked if this is turned on and `ingress.enabled` is `false`

File: stable/burrow/README.md [hunk 13]
```diff
@@ -1,164 +1,183 @@
...
+| `rpcInfo.ingress.pathLeader` | - | `"/"` |
+| `rpcInfo.ingress.annotations` | extra annotations | `` |
+| `rpcInfo.ingress.hosts` | - | `[]` |
+| `rpcInfo.ingress.tls` | - | `` |
+| `rpcMetrics.enabled` | enable RPC service | `true` |
+| `rpcMetrics.port` | RPC port | `9102` |
+| `rpcMetrics.path` | http endpoint | `"/metrics"` |
+| `rpcMetrics.blockSampleSize` | number of previous blocks to fetch | `100` |
```
Comment (compleatang) [inline stable/burrow/README.md]: `number of previous blocks to utilize in calculating the histograms and summaries which are sent to prometheus`

File: stable/burrow/README.md [hunk 14]
```diff
@@ -1,164 +1,183 @@
...
+| `nodeSelector` | node labels for pod assignment | `{}` |
 
 Specify each parameter using the `--set key=value[,key=value]` argument to `helm install`. For example,
 
-```console
-$ helm install stable/burrow --name my-release \
-  --set=image.tag=0.16.0,resources.limits.cpu=200m
+```bash
+helm install stable/burrow --name my-release \
+  --set=image.tag=0.23.1,resources.limits.cpu=200m
```
Comment (compleatang) [inline stable/burrow/README.md]: let's add the extra values file here as well.

File: stable/burrow/README.md [hunk 15]
```diff
@@ -1,164 +1,183 @@
...
-### Configuration of the Ports
+### Port Configuration
 
 Burrow utilizes three different ports by default:
 
-* `peer`: Burrow's peer port is the port on which the p2p communication within the blockchain network is conducted. The peer port is utilized by burrow's consensus engine (which is the [Tendermint](https://github.com/tendermint/tendermint) engine) to perform bilateral gossiping communication.
-* `rpc`: Burrow's rpc port is the port on which remote procedures are conducted. It is utilized by tools such as, e.g., the [Monax](https://github.com/monax/monax) tools which can be utilized to deploy smart contracts and perform other functions.
-* `api`: Burrow's api port is the port on which javascript libraries interact with the chain utilizing websockets. It is generally utilized by tools.
+* `Peer`: Burrow's peer port is used for P2P communication within the blockchain network as part of the consensus engine ([Tendermint](https://github.com/tendermint/tendermint)) to perform bilateral gossiping communication.
+* `RPC`: Burrow's RPC port is used for conducting remote procedures.
```
Comment (compleatang) [inline stable/burrow/README.md]: for consistency let's `s/RPC/Info/g` -- this port is used for gaining information from a running burrow node over http calls with json returns.

File: stable/burrow/README.md [hunk 16]
```diff
@@ -1,164 +1,183 @@
...
+* `Peer`: Peer ports are **only** opened within the cluster. By default, there is no P2P communication exposed to the general internet. Each node within the cluster has its own distinct peer service built by the chart which utilizes a `ClusterIP` service type.
+* `RPC`: The RPC port is **only** opened within the cluster. By default, there is no RPC communication exposed to the general internet. There is **one** RPC service built by the chart which utilizes a `ClusterIP` service type. The default RPC service used by the chart is strongly linked to node number `000` and is not load balanced across the nodes by default so as to reduce any challenges with tooling that conduct long-polling after sending transactions. The chart offers an ingress which is connected to the `RPC` service, but this is `disabled` by default.
```
Comment (compleatang) [inline stable/burrow/README.md]: see above (`s/RPC/Info/g`)

**Review by compleatang — COMMENTED** [review]
Nice work! Comments added herein.

**Comment by gregdhill** [comment]
lgtm

**Pull request merged**